### PR TITLE
Integrates questions API with front

### LINKED ISSUE #531: Funcionalidade de Perguntas Frequentes da tecnologia
## Feature Description

Será criada uma aba nova nos detalhes da tecnologia chamada PERGUNTAS FREQUENTES, no qual os usuários poderão fazer perguntas ao pesquisador responsável pela tecnologia. Terá as seguintes características:

VISÃO DO USUÁRIO
- Apresentar uma lista de perguntas e a resposta abaixo.
- Apresentar um botão FAZER UMA PERGUNTA
- Ao clicar no botão FAZER UMA PERGUNTA levará a um modal com um campo de pergunta (textarea) e o botão enviar
- Ao enviar, o pesquisador responsável recebe a notificação por email da pergunta

VISÃO DO PESQUISADOR
- Na área do pesquisador incluir mais um item de menu chamado PERGUNTAS (incluir um badget com a quantidade de perguntas SEM RESPOSTA)
- No Menu PERGUNTAS apresentar uma lista com a pergunta, a tecnologia vinculada, o usuário que fez a pergunta, o status (SEM RESPOSTA, RESPONDIDO, DESATIVADA), e ações (RESPONDER, DESATIVAR)
- Ao clicar em RESPONDER aparecer um modal com um campo textarea de resposta e o botão PUBLICAR RESPOSTA.
- Ao clicar em PUBLICAR RESPOSTA registrar a resposta e alterar o status para RESPONDIDO e apresentar a mensagem que a resposta que foi publicada com sucesso. 
- Ao clicar em DESATIVAR, alterar o status da pergunta para DESATIVADO e apresentar a mensagem que a pergunta não está mais publicada para os usuários visualizarem.

---------------

## Acceptance criteria

* 
- Qualquer usuário cadastrado e logado poderá fazer uma pergunta
- Qualquer usuário pesquisador responsável pela pergunta poderá registrar uma resposta

## Implementation Brief

*

### PR DESCRIPTION
**Pull request by calaca**
## Summary

Resolves #531 

Peek 2020-12-04 16-11
PS: there are only answred questions being listed now.

Peek 2020-12-04 16-15

## QA Steps
1. See details of a technology that has no questions
2. Check that there's a message saying so
3. Send a new question
4. See the question listed bellow
***
1. See details of a technology while logged out
2. Check that you can't send a new question
***
1. Go to the questions page at your profile
2. Check that you can see info about the question, refuse or answer it

## Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code passes the linting.
- [x] My code has proper inline documentation.
- [x] I have manually tested this PR.

### CODE REVIEW

File: packages/web/components/Technology/Details/Tabs/FAQ/index.js [hunk 1]
```diff
@@ -1,69 +1,96 @@
...
+			toast.error('Houve um erro ao enviar sua pergunta');
+		}
+
+		form.reset();
 	};
 
 	const loadMoreQuestions = () => {
-		toast.error('Carregar mais 10 perguntas e respostas');
+		setItemsCount(itemsCount + 5);
+		mutate('get-technology-answered-questions');
```
Comment (luizeboli) [inline packages/web/components/Technology/Details/Tabs/FAQ/index.js:59]: You don't need to pass key here as this mutate is already bound to that query

See https://swr.vercel.app/docs/mutation#bound-mutate

**Review by luizeboli — CHANGES_REQUESTED** [review]
Awesome, great job :tada: 

I made one comment here in addition to the video I sent you in Slack

File: packages/web/components/UserProfile/UserProfile.js [hunk 2]
```diff
@@ -9,12 +10,21 @@ import BeAReviewerButton from './BeAReviewerButton';
...
 	const { user } = useAuth();
 	const { t } = useTranslation(['profile']);
 	const router = useRouter();
 
+	const {
+		data: { data },
+	} = useSWR(['get-user-unanswered-questions-count'], () => getUserUnansweredQuestions(), {
+		initialData: [],
+		revalidateOnMount: true,
+		revalidateOnFocus: true,
```
Comment (luizeboli) [inline packages/web/components/UserProfile/UserProfile.js:25]: Focus revalidation is true by default 😄

**Review by luizeboli — APPROVED** [review]
Nice work done 🥳

File: packages/web/components/Technology/Details/Tabs/FAQ/index.js [hunk 3]
```diff
@@ -1,69 +1,96 @@
...
+				)}
+
+				{questions?.length ? (
+					<Loading loading={isValidating}>
+						<S.LastQuestions>
+							<S.LastQuestionsTitle>Últimas realizadas</S.LastQuestionsTitle>
+							{questions?.map((question) => (
+								<Question key={question.id} question={question} />
+							))}
+							<S.LoadMoreButton
```
Comment (lcnogueira) [inline packages/web/components/Technology/Details/Tabs/FAQ/index.js:86]: This load more feature is awesome! 👏

**Review by lcnogueira — APPROVED** [review]
Can confirm it's working as expected. Great job, @calaca !

**Pull request merged**